docs/DafnyRef/Specifications.md

```diff
@@ -844,3 +844,58 @@ add:
    requires Valid()
    reads Repr
 ```
+
+## 7.8. Well-formedness of specifications {#sec-well-formedness}
+
+Dafny verifies that the requires and ensures clauses make sense by themselves, independently of the body.
+Hence, by declaring the following method:
+
+```dafny
+method Test(a: array<int>) returns (j: int)
+  requires a.Length >= 1
+  ensures a.Length % 2 == 0 ==> j >= 10 / a.Length
+{
+  j := 20;
+  var divisor := a.Length;
+  if divisor % 2 == 0 {
+    j := j / divisor;
+  }
+}
+```
+
+Dafny will split the verification in two [assertion batches](#sec-assertion-batches)
+that will roughly look like the following lemmas:
+
+```dafny
+lemma Test_WellFormed(a: array<int>)
+{
+  assert a != null;       // for the `requires a.Length >= 1``
```

Comment:
No it does not because it's in triple backticks anyway. But I see that in many places where I intended only one backtick, it writes two, let me fix that.